ariadne/cardano/src/Ariadne/Wallet/Cardano/Kernel/Addresses.hs

```diff
@@ -168,15 +167,14 @@ createHdSeqAddress passphrase esk accId chain pw = runExceptT $ go 0
             (CreateAddressHdSeqGenerationFailed accId)
             mbAddr
 
-        let hdAddress = initHdAddress hdAddressId (InDb newAddress) firstCheckpoint
-        res <- liftIO $ update db (CreateHdAddress hdAddress)
+        res <- liftIO $ update db (CreateHdAddress hdAddressId newAddress firstCheckpoint)
```

Comment:
You should't pass `firstCheckpoint` there. `createHdAddress` can do it internally.